weldx/tests/transformations/test_cs_manager.py

```diff
@@ -88,56 +88,50 @@ def test_init():
-#  how 2 coordinates are stored in the Dataset. This lets the coordinate comparison
-#  fail.
-csm_acs = CSM("root")
-time = pd.DatetimeIndex(["2000-01-01", "2000-01-04"])
-# lcs_2_acs = LCS(coordinates=[[0, -1, -2], [8, 2, 7]], time=time)
 
-
-@pytest.mark.parametrize(
-    "name , parent, lcs, child_in_parent, exp_num_cs",
-    [
-        ("lcs1", "root", LCS(coordinates=[0, 1, 2]), True, 2),
-        ("lcs2", "root", LCS(coordinates=[0, -1, -2]), False, 3),
-        ("lcs3", "lcs2", LCS(r_mat_y(1 / 2), [1, 2, 3]), True, 4),
-        ("lcs3", "lcs2", LCS(coordinates=[-1, -2, -3]), True, 4),
-        ("lcs2", "lcs3", LCS(coordinates=[-1, -2, -3]), False, 4),
-        ("lcs2", "lcs3", LCS(coordinates=[-1, -2, -3]), True, 4),
-        ("lcs4", "lcs2", LCS(coordinates=[0, 1, 2]), True, 5),
-        ("lcs4", "lcs2", LCS(r_mat_y(1 / 2), [1, 2, 3]), True, 5),
-        ("lcs5", "lcs1", LCS(r_mat_y(3 / 2), [2, 3, 1]), True, 6),
-        (
-            "lcs5",
-            "lcs1",
-            LCS(
-                None,
-                TimeSeries(MathematicalExpression("a*t", dict(a=Q_(1, "1/s")))),
-            ),
-            True,
-            6,
-        ),
-    ],
-)
-def test_add_coordinate_system(name, parent, lcs, child_in_parent, exp_num_cs):
+def test_add_cs():
     """Test the 'add_cs' function."""
-    csm = csm_acs
-    csm.add_cs(name, parent, lcs, child_in_parent)
-
+    csm = CSM("r")
+    ts = TimeSeries(MathematicalExpression("a*t", dict(a=Q_("1/s"))))
+
+    lcs_data = [
+        ("a", "r", LCS(coordinates=[0, 1, 2]), True),
+        ("b", "r", LCS(coordinates=[0, -1, -2]), False),
+        ("b", "r", LCS(coordinates=[[0, -1, -2], [8, 2, 7]], time=["1s", "2s"]), False),
+        ("c", "b", LCS(r_mat_y(1 / 2), [1, 2, 3]), True),
+        ("c", "b", LCS(coordinates=[-1, -2, -3]), True),
+        ("b", "c", LCS(coordinates=[-1, -2, -3]), False),
+        ("b", "c", LCS(coordinates=[-1, -2, -3]), True),
+        ("d", "b", LCS(coordinates=[0, 1, 2]), True),
+        ("d", "b", LCS(r_mat_y(1 / 2), [1, 2, 3]), True),
+        ("e", "a", LCS(r_mat_y(3 / 2), [2, 3, 1]), True),
+        ("e", "a", LCS(coordinates=ts), True),
+    ]
+    exp_num_cs = 1
     assert csm.number_of_coordinate_systems == exp_num_cs
-    if child_in_parent:
-        assert csm.get_cs(name, parent) == lcs
-        if not isinstance(lcs.coordinates, TimeSeries):
-            assert csm.get_cs(parent, name) == lcs.invert()
-    else:
-        if not isinstance(lcs.coordinates, TimeSeries):
-            assert csm.get_cs(name, parent) == lcs.invert()
-        assert csm.get_cs(parent, name) == lcs
+
+    for d in lcs_data:
+        name = d[0]
+        parent = d[1]
+        lcs = d[2]
+        child_in_parent = d[3]
+
+        if name not in csm.coordinate_system_names:
+            exp_num_cs += 1
+
+        csm.add_cs(name, parent, lcs, child_in_parent)
+
+        assert csm.number_of_coordinate_systems == exp_num_cs
```

Comment:
Thanks for fixing this. I suggest that you put a common assert message indicating the former input parameters. This is of course only interesting in case of a failing test, but eventually one does not need to fire up a debugger to know which case is giving trouble.

Reply:
Good point. I will add a message.

Reply:
Added a short message to all assertions. Not beautiful, but should do the trick.